examples/3d/elixir_advection_amr_p4est_unstructured_curved.jl

```diff
@@ -0,0 +1,105 @@
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the linear advection equation
+
+advectionvelocity = (0.2, -0.7, 0.5)
+equations = LinearScalarAdvectionEquation3D(advectionvelocity)
+
+solver = DGSEM(polydeg=3, surface_flux=flux_lax_friedrichs)
+
+initial_condition = initial_condition_gauss
+boundary_condition = BoundaryConditionDirichlet(initial_condition)
+
+boundary_conditions = Dict(
+  :all => boundary_condition
+)
+
+# Mapping as described in https://arxiv.org/abs/2012.12040, but with less warping.
+# The original mapping applied to this unstructured mesh creates extreme angles,
+# which require a high resolution for proper results.
```

Comment:
This problem specifically occurs when I evaluate the geometry with a low resolution (on the trees), and then refine the mesh without changing the geometry. I'm not sure who's in the Slack channel. Should I post my findings here as well?

Reply:
Good point. Maybe just create an issue with your findings - even if there's nothing to do but acknowledge that the resolution must be sufficiently high to not create degenerate elements, it would be a nice documentation of all the work you've put in. Then you can just resolve this conversation an move on.

Reply:
See #660.

Reply:
Yes, I've seen similar issues on coarse resolutions in 2D.